configure.ac

```diff
@@ -6340,6 +6340,21 @@ AS_VAR_IF([with_readline], [no], [
     # in readline as well as newer editline (April 2023)
     AC_CHECK_TYPES([rl_compdisp_func_t], [], [], [readline_includes])
 
+    # Some editline versions declare rl_startup_hook as taking no args, others
+    # declare it as taking 2.
+    AC_CACHE_CHECK([if rl_startup_hook takes arguments], [ac_cv_readline_rl_startup_hook_takes_args], [
+        AC_COMPILE_IFELSE(
+            [AC_LANG_PROGRAM([readline_includes]
+                [extern int test_hook_func(const char *text, int state);],
+                [rl_startup_hook=test_hook_func;])],
+            [ac_cv_readline_rl_startup_hook_takes_args=yes],
+            [ac_cv_readline_rl_startup_hook_takes_args=no]
+        )
+    ])
+    AS_VAR_IF([ac_cv_readline_rl_startup_hook_takes_args], [yes], [
+      AC_DEFINE([Py_RL_STARTUP_HOOK_TAKES_ARGS], [1], [Define if rl_startup_hook takes arguments])
```

Comment:
I am not sure about the naming convention in ac. Most names we define here have `HAVE_` prefix, for example `HAVE_RL_APPEND_HISTORY`

Maybe we should use a similar name?

Reply:
`HAVE_` is normally used to confirm the existence of an API. In this case we want to confirm the signature of an API. I think the proposed name is fine.

More importantly, I think we should make sure that new additions to pyconfig.h.in follow the PEP-7 guidelines. The existing inconsistencies have been discussed, and it is something we want to clean up. (There has already been cleanups for incorrectly named defines and macros). Let's not add more inconsistencies :)

Reply:
See also:

- #105813